tasks/generate-history/createSarcophagi.ts

```diff
@@ -0,0 +1,170 @@
+  const embalmerFacet = await hre.ethers.getContractAt("EmbalmerFacet", diamond.address);
+  const sarcoToken = await hre.ethers.getContract("SarcoTokenMock");
+
+  console.log();
+  console.log("Creating sarcophagi...");
+  const sarcophagiData = [];
+  for (let i = 0; i < sarcophagusCount; i++) {
+    // Define the sarcohpagus parameters
+    const fakeSarcoId = hexlify(randomBytes(32));
+
+    // The number of archaeologists on the sarcophagus, maxes out at maxArchaeologistOnSarcophagus
+    const selectedArchaeologistCount = range(
+      1,
+      archaeologistSigners.length > maxArchaeologistOnSarcophagus
+        ? maxArchaeologistOnSarcophagus
+        : archaeologistSigners.length
+    );
+    const minShards = range(1, selectedArchaeologistCount);
+    const name = "Sarcophagus Name";
+
+    // Pick random addresses from the list of archaeologist addresses to be added to the
+    // sarcophagus. Do not allow duplicates.
+    // Shuffle and slice the array
+    const selectedArchaeologistSigners = archaeologistSigners
+      .sort(() => 0.5 - Math.random())
+      .slice(0, selectedArchaeologistCount);
+
+    // Create the shards using Shamirs Secret Sharing
+    const outerLayerPrivateKey = "ce6cb1ae13d79a053daba0e960411eba8648b7f7e81c196fd6b36980ce3b3419";
+    const shards: Uint8Array[] = split(outerLayerPrivateKey, {
+      shares: selectedArchaeologistCount,
+      threshold: minShards,
+    });
+
+    // Assign a shard to each archaeologist
+    const shardsMap: { [key: string]: Uint8Array } = {};
+    for (let j = 0; j < selectedArchaeologistSigners.length; j++) {
+      const archaeologistSigner = selectedArchaeologistSigners[j];
+      const shard = shards[j];
+      shardsMap[archaeologistSigner.address] = shard;
+    }
+
+    // Create the archaeologist objects to be passed in to the createSarcophagus function
+    const archaeologists = [];
+    for (let i = 0; i < selectedArchaeologistCount; i++) {
+      const archaeologistSigner = selectedArchaeologistSigners[i];
+
+      // Double hash this archaeologist's shard
+      const doubleHashedShard = keccak256(keccak256(shardsMap[archaeologistSigner.address]));
+      // arweaveTxId, unencryptedShardDoubleHash, agreedMaximumRewrapInterval, diggingFee, timestamp
+      try {
+        // The archaeologist signs the double hashed shard and a fake arweave transaction id
+        const signature = await signHre(
+          hre,
+          selectedArchaeologistSigners[i],
+          [
+            fakeShardTxId,
+            doubleHashedShard,
+            maximumRewrapInterval.toString(),
+            diggingFee.toString(),
+            timestamp.toString(),
+          ],
+          ["string", "bytes32", "uint256", "uint256", "uint256"]
+        );
+        // const signature = await signHre(
+        //   hre,
+        //   selectedArchaeologistSigners[i],
+        //   [doubleHashedShard, fakeShardTxId],
+        //   ["bytes32", "string"]
+        // );
```

Comment:
can this signature block go bye-byte b/c there is the one above?

Reply:
Yeah forgot to take this out